client/app/react-components/QueryEditor.js

```diff
@@ -0,0 +1,229 @@
+import React from 'react';
```

Comment:
Please change the filename extension to `.jsx`. Also as we have a dedicated extension for React component, I think we can move them back into `app/component` out of their `app/react-component` exile :)